sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -574,6 +574,20 @@ public void setSessionFactory(SessionFactory sessionFactory) {
   }
 
   @Override
+  public Collection<Identity> getIdentitiesByExternalNameAndProject(String externalName, Long projectId) throws IOException {
+    if (isStringEmptyOrNull(externalName)) return Collections.emptySet();
+    if (projectId == null) throw new IllegalArgumentException("Must provide a projectId in search");
+    String str = DbUtils.convertStringToSearchQuery(externalName);
+    Criteria criteria = currentSession().createCriteria(IdentityImpl.class);
+    criteria.add(Restrictions.eq("project.id", projectId));
+    criteria.add(Restrictions.or(Restrictions.ilike("externalName", str), Restrictions.ilike("alias", str)));
+    @SuppressWarnings("unchecked")
+    Collection<Identity> records = criteria.list();
+    // return without fetching all associated SQL items
```

Comment:
Ah. Okay. Please make a reference to the `fetchSqlStore` function that is not being run.